homeassistant/components/prusalink/button.py

```diff
@@ -15,14 +16,14 @@
 
 from . import DOMAIN, PrusaLinkEntity, PrusaLinkUpdateCoordinator
 
-T = TypeVar("T", PrinterInfo, JobInfo)
+T = TypeVar("T", PrinterStatus, LegacyPrinterStatus, JobInfo)
 
 
 @dataclass(frozen=True)
 class PrusaLinkButtonEntityDescriptionMixin(Generic[T]):
     """Mixin for required keys."""
 
-    press_fn: Callable[[PrusaLink], Coroutine[Any, Any, None]]
+    press_fn: Callable[[PrusaLink, PrinterStatus], Coroutine[Any, Any, None]]
```

Comment:
```suggestion
    press_fn: Callable[[PrusaLink], Callable[[int], Coroutine[Any, Any, None]]]
```

As all function only need the job id, we can just return the function

Reply:
Good idea! Thanks